aws-rollback.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python3
+
+import json
+import subprocess
+import sys
+
+TARGET_TAG = 'latest'
+ECS_CLUSTER = 'rust-ecs-prod'
+
+def main():
+    """CLI entrypoint of the program."""
+    if len(sys.argv) < 2:
+        usage()
+    repository_name = sys.argv[1]
+    images = get_images(repository_name)
+    while True:
+        selected_image_index = let_user_pick_image(images)
+        if selected_image_index == -1:
+            exit(0)
+
+        if selected_image_index is not None:
+            break
+
+    image = images[selected_image_index]
+    if image["imageTag"] == TARGET_TAG:
+        err(f"selected image already tagged as {TARGET_TAG}")
+
+    eprint(f"selected option: {image}")
+    manifest = get_image_manifest(repository_name, image["imageTag"])
+    retag_image(repository_name,manifest)
+    eprint(f"image {image} retaged as '{TARGET_TAG}'")
+    if can_redeploy(repository_name):
+        redeployed = force_redeploy()
+        if redeployed:
+            eprint(f"successfully rollback and re-deploy")
+
+def let_user_pick_image(images):
+    print("Please choosean image to rollback:")
+    for idx, image in enumerate(images):
+        print("{}) {}".format(idx+1,image["imageTag"]))
```

Comment:
In our repository, only the `latest` image is tagged, while all the past ones are untagged. It would be nice if the prompt looked something like this (the list is sorted from the most recently pushed image to the oldest):

```
Please choose an image to rollback:
2) 2020-04-13 12:02:39 (latest)
3) 2020-04-11 00:25:25
4) 2020-04-06 12:02:30 (other-tag)
5) 2020-04-01 13:15:24
```

You can use the `aws ecr describe-images` command instead of `aws ecr list-images` to fetch the additional information. Also, since not all the images are tagged you'll need to use digests to identify images.